src/attrib.d

```diff
@@ -409,17 +410,26 @@ public:
 
     override void setScope(Scope* sc)
     {
+        Scope* scx = sc.push();
+        scx.depdecl = this;
+        StorageClassDeclaration.setScope(scx);
+        scx.pop();
+    }
```

Comment:
When you save a scope `scx` by `setScope(scx)`, the scx should be an *enclosing scope* of this `DeprecatedDeclaration`, ant it should be equivalent with the scope object which will be passed to `semantic(Scope* sc)`, `semantic2(Scope* sc)`, and others.

However, this code saves a new scope which is set the `depdecl` field to `this`. It will mean the enclosing deprecation attribute of `this` is `this` - there's recursive structure! Of course it's not good.

Instead, you can override `newScope` member function to construct a new scope for the members of `DeprecatedDeclaration`.